app/components/DataInput/index.js

```diff
@@ -24,18 +24,17 @@ class DataInput extends AppComponent {
     });
   }
 
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this._submitData = this._submitData.bind(this);
     this._loadSampleData = this._loadSampleData.bind(this);
     this._setSampleDataSet = this._setSampleDataSet.bind(this);
-    this._beforeNextStep = this._beforeNextStep.bind(this);
     this._nextCallback = this._nextCallback.bind(this);
     this._handleInputBlur = this._handleInputBlur.bind(this);
     this._handleInputChange = this._handleInputChange.bind(this);
 
     this.state = {
-      rawData: '',
+      rawData: props.rawData,
```

Comment:
A general rule that I follow is that I wait until object property access happens 2+ times with the same object, before I destructure.

The short hand property set is nice, but it's technically an extra line of code here.